{"hierarchy":{"paths":[["doc:\/\/flowkit.FlowKit\/documentation\/FlowKit","doc:\/\/flowkit.FlowKit\/documentation\/FlowKit\/AsyncThrowingSubject"]]},"identifier":{"url":"doc:\/\/flowkit.FlowKit\/documentation\/FlowKit\/AsyncThrowingSubject\/makeAsyncIterator()","interfaceLanguage":"swift"},"abstract":[{"type":"text","text":"Creates and returns an asynchronous iterator for the sequence."}],"primaryContentSections":[{"declarations":[{"platforms":["iOS"],"languages":["swift"],"tokens":[{"text":"func","kind":"keyword"},{"kind":"text","text":" "},{"text":"makeAsyncIterator","kind":"identifier"},{"kind":"text","text":"() -> "},{"text":"AsyncThrowingStream","kind":"typeIdentifier","preciseIdentifier":"s:Scs"},{"text":"<","kind":"text"},{"kind":"typeIdentifier","text":"Element","preciseIdentifier":"s:7FlowKit20AsyncThrowingSubjectV7Elementxmfp"},{"text":", ","kind":"text"},{"kind":"typeIdentifier","identifier":"doc:\/\/flowkit.FlowKit\/documentation\/FlowKit\/AsyncThrowingSubject\/Failure","preciseIdentifier":"s:7FlowKit20AsyncThrowingSubjectV7Failurea","text":"Failure"},{"kind":"text","text":">."},{"preciseIdentifier":"s:Scs8IteratorV","kind":"typeIdentifier","text":"Iterator"}]}],"kind":"declarations"},{"kind":"content","content":[{"level":2,"anchor":"return-value","type":"heading","text":"Return Value"},{"type":"paragraph","inlineContent":[{"type":"text","text":"An iterator that produces elements asynchronously."}]}]}],"variants":[{"traits":[{"interfaceLanguage":"swift"}],"paths":["\/documentation\/flowkit\/asyncthrowingsubject\/makeasynciterator()"]}],"kind":"symbol","metadata":{"title":"makeAsyncIterator()","role":"symbol","externalID":"s:7FlowKit20AsyncThrowingSubjectV04makeC8IteratorScs0G0Vyxs5Error_p_GyF","fragments":[{"text":"func","kind":"keyword"},{"kind":"text","text":" "},{"kind":"identifier","text":"makeAsyncIterator"},{"text":"() -> ","kind":"text"},{"text":"AsyncThrowingStream","kind":"typeIdentifier","preciseIdentifier":"s:Scs"},{"kind":"text","text":"<"},{"kind":"typeIdentifier","preciseIdentifier":"s:7FlowKit20AsyncThrowingSubjectV7Elementxmfp","text":"Element"},{"text":", ","kind":"text"},{"kind":"typeIdentifier","preciseIdentifier":"s:7FlowKit20AsyncThrowingSubjectV7Failurea","text":"Failure"},{"text":">.","kind":"text"},{"preciseIdentifier":"s:Scs8IteratorV","text":"Iterator","kind":"typeIdentifier"}],"symbolKind":"method","modules":[{"name":"FlowKit"}],"roleHeading":"Instance Method"},"sections":[],"schemaVersion":{"patch":0,"minor":3,"major":0},"references":{"doc://flowkit.FlowKit/documentation/FlowKit/AsyncThrowingSubject":{"fragments":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"kind":"identifier","text":"AsyncThrowingSubject"}],"url":"\/documentation\/flowkit\/asyncthrowingsubject","type":"topic","abstract":[{"text":"A subject that allows for asynchronously sending values or terminating with an error.","type":"text"},{"type":"text","text":" "},{"type":"codeVoice","code":"AsyncThrowingSubject"},{"text":" acts as a bridge between imperative code and asynchronous code, enabling","type":"text"},{"type":"text","text":" "},{"text":"you to send values to an ","type":"text"},{"type":"codeVoice","code":"AsyncSequence"},{"text":" consumer while handling potential errors gracefully.","type":"text"}],"kind":"symbol","navigatorTitle":[{"kind":"identifier","text":"AsyncThrowingSubject"}],"title":"AsyncThrowingSubject","role":"symbol","identifier":"doc:\/\/flowkit.FlowKit\/documentation\/FlowKit\/AsyncThrowingSubject"},"doc://flowkit.FlowKit/documentation/FlowKit/AsyncThrowingSubject/makeAsyncIterator()":{"abstract":[{"type":"text","text":"Creates and returns an asynchronous iterator for the sequence."}],"identifier":"doc:\/\/flowkit.FlowKit\/documentation\/FlowKit\/AsyncThrowingSubject\/makeAsyncIterator()","title":"makeAsyncIterator()","kind":"symbol","type":"topic","url":"\/documentation\/flowkit\/asyncthrowingsubject\/makeasynciterator()","role":"symbol","fragments":[{"kind":"keyword","text":"func"},{"text":" ","kind":"text"},{"text":"makeAsyncIterator","kind":"identifier"},{"kind":"text","text":"() -> "},{"text":"AsyncThrowingStream","preciseIdentifier":"s:Scs","kind":"typeIdentifier"},{"kind":"text","text":"<"},{"kind":"typeIdentifier","preciseIdentifier":"s:7FlowKit20AsyncThrowingSubjectV7Elementxmfp","text":"Element"},{"text":", ","kind":"text"},{"kind":"typeIdentifier","text":"Failure","preciseIdentifier":"s:7FlowKit20AsyncThrowingSubjectV7Failurea"},{"text":">.","kind":"text"},{"text":"Iterator","kind":"typeIdentifier","preciseIdentifier":"s:Scs8IteratorV"}]},"doc://flowkit.FlowKit/documentation/FlowKit/AsyncThrowingSubject/Failure":{"abstract":[{"type":"text","text":"The type of error that can terminate the sequence."}],"identifier":"doc:\/\/flowkit.FlowKit\/documentation\/FlowKit\/AsyncThrowingSubject\/Failure","title":"AsyncThrowingSubject.Failure","kind":"symbol","type":"topic","url":"\/documentation\/flowkit\/asyncthrowingsubject\/failure","role":"symbol","navigatorTitle":[{"kind":"identifier","text":"Failure"}],"fragments":[{"text":"typealias","kind":"keyword"},{"kind":"text","text":" "},{"text":"Failure","kind":"identifier"}]},"doc://flowkit.FlowKit/documentation/FlowKit":{"abstract":[],"kind":"symbol","url":"\/documentation\/flowkit","title":"FlowKit","identifier":"doc:\/\/flowkit.FlowKit\/documentation\/FlowKit","type":"topic","role":"collection"}}}